Sprinkle of Yeezus/Settings.swift

```diff
@@ -11,52 +11,68 @@ import UIKit
 import UserNotifications
 
 class SettingsPage: UIViewController {
-    
 
-    @IBOutlet weak var InfoLabel: UILabel!
-    @IBOutlet weak var DescButton: UIButton!
-    @IBOutlet weak var DescLabel: UILabel!
-    @IBOutlet weak var Switch: UISwitch!
+    @IBOutlet private weak var InfoLabel: UILabel!
+    @IBOutlet private weak var DescButton: UIButton!
+    @IBOutlet private weak var DescLabel: UILabel!
+    @IBOutlet private weak var Switch: UISwitch!
+    @IBOutlet private weak var TimePicked: UIDatePicker!
     
-    var notificationsOn = Bool()
+    private var sprinkleTimePicked = Date()
+    private let notificationCenter = UNUserNotificationCenter.current()
     
-    @IBAction func NotificationSwitch(_ sender: UISwitch) {
-        if Switch.isOn == true{
-            UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .sound], completionHandler: {didAllow, error in})
-            
-            DescLabel.isHidden = false
-            DescButton.isHidden = false
-            
-            notificationsOn = true
-        } else {
-            DescLabel.isHidden = true
-            DescButton.isHidden = true
-            notificationsOn = false
-        }
+    // MARK: - Lifecycle
+    override func viewDidLoad() {
+        sprinkleNotifications(sprinkleList)
     }
     
-    @IBOutlet weak var TimePicked: UIDatePicker!
-    var sprinkleTimePicked = Date()
-    @IBAction func TimePicker(_ sender: UIDatePicker){
-        sprinkleTimePicked = TimePicked.date
+    // MARK: - IBAction
+    @IBAction private func NotificationSwitch(_ sender: UISwitch) {
+        if sender.isOn {
+            notificationCenter.requestAuthorization(options: [.alert, .sound], completionHandler: { didAllow, error in
+                DispatchQueue.main.async {
+                    sender.isOn = didAllow
+                    self.updateUI()
+                }
+            })
+        }
+        
+        updateUI()
     }
     
-    func sprinkleNotifications(_ sprinkleList: Array<Sprinkle>) {
-        if notificationsOn == true{
```

Comment:
In cases like this, the value of `Switch.isOn` should be used as the conditional. `Switch.isOn` is the same as `Switch.isOn == true`. You'll see how I refactored the rest of this function below.